app/controllers/shops_controller.rb

```diff
@@ -54,18 +54,21 @@ def index_appointment
   end
 
   def change_status
+    @appointment = Appointment.find(params[:appointment])
     @company = Company.find(params[:company_id])
     authorize! :show, @company
     @shop = Shop.find(params[:shop_id])
     authorize! :show, @shop
     @appointment = Appointment.find(params[:appointment])
     if @appointment.status == 'Booked'
       @appointment.update(status: 'Ready for pickup')
+      AppointmentMailer.status_email(@appointment).deliver_now
```

Comment:
use a better name for the email. one that suggests what the email is for. Don't forget to rename the template

Reply:
name has been changed to change_appointment_status_email